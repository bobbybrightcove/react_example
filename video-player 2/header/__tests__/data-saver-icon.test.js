import React from 'react';
import renderer from 'react-test-renderer';
import DataSaverIcon from '../data-saver-icon';


describe('DataSaverIcon', () => {
  const player = {};

  it('renders according to valid snapshot', () => {
    const component = renderer.create(
      <DataSaverIcon
        player={player}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change icon on mouse enter/leave', () => {
    const component = renderer.create(
      <DataSaverIcon
        player={player}
      />
    );

    // onMouseEnter
    component.toJSON().children[0].props.onMouseEnter();

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // onMouseLeave
    tree.children[0].props.onMouseLeave();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change to HD icon if HD is selected', () => {
    const component = renderer.create(
      <DataSaverIcon
        player={player}
        selected={{ label: 'hd' }}
      />
    );

    // onMouseEnter
    component.toJSON().children[0].props.onMouseEnter();

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // onMouseLeave
    tree.children[0].props.onMouseLeave();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change to SD icon if SD is selected', () => {
    const component = renderer.create(
      <DataSaverIcon
        player={player}
        selected={{ label: 'sd' }}
      />
    );

    // onMouseEnter
    component.toJSON().children[0].props.onMouseEnter();

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // onMouseLeave
    tree.children[0].props.onMouseLeave();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change to SAVER icon if SAVER is selected', () => {
    const component = renderer.create(
      <DataSaverIcon
        player={player}
        selected={{ label: 'saver' }}
      />
    );

    // onMouseEnter
    component.toJSON().children[0].props.onMouseEnter();

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // onMouseLeave
    tree.children[0].props.onMouseLeave();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
