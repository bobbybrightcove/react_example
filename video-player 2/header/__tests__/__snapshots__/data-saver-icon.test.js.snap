// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSaverIcon renders according to valid snapshot 1`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-hd.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change icon on mouse enter/leave 1`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-hd-hover.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change icon on mouse enter/leave 2`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-hd.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change to HD icon if HD is selected 1`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-hd-hover.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change to HD icon if HD is selected 2`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-hd.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change to SAVER icon if SAVER is selected 1`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-saver-hover.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change to SAVER icon if SAVER is selected 2`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-saver.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change to SD icon if SD is selected 1`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-sd-hover.svg"
    />
  </div>
</div>
`;

exports[`DataSaverIcon should change to SD icon if SD is selected 2`] = `
<div
  className="dataSaverWrapper"
>
  <div
    className="dataSaverIconBox"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Data Saver"
      src="data-saver-sd.svg"
    />
  </div>
</div>
`;
